Assets/CarriageManager.cs

```diff
@@ -0,0 +1,143 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarriageManager : MonoBehaviour
+{
+    public List<GameObject> carriagePrefabs;
+
+    private List<GameObject> carriagesToSpawn;
+    [SerializeField] private List<GameObject> spawnedCarriages;
+    [SerializeField] private CarriageData currentCarriage, oldCarriage;
+
+    //SPAWNING
+    [SerializeField] private Transform firstSpawn;
+    [SerializeField] private Transform nextSpawnPos,prevSpawnPos;
+
+    //MOVING
+    private void Awake()
+    {
+    }
+    private void Start()
+    {
+        InitialiseCarriages();
+    }
+
+    private void InitialiseCarriages()
+    {
+        //ALGORITHM CAN BE ALTERED TO BE RANDOMISED ON EVERY NEW PLAYTHROUGH OR KEPT AS IS
+
+        carriagesToSpawn = new List<GameObject>(carriagePrefabs.Count);
+        foreach (GameObject _carriage in carriagePrefabs)
+            carriagesToSpawn.Add(_carriage);
+
+
+        //PREPARE SPAWN FIRST CARRIAGE
+        nextSpawnPos = firstSpawn;
+
+        //NAMING
+        int _count = 0;
```

Comment:
could you tell me why so many variable names start with an underscore? I feel like this is a unity convention that I am not familiar with, apologies

Reply:
It just looks a little odd to me right now

Reply:
underscored variables refer to local variables. It was a convention taught to me by a web developer.
Helps differentiate local variables and stored variables.